src/domain/templates/templates.types.ts

```diff
@@ -0,0 +1,42 @@
+import { z } from 'zod';
+
+const baseFields = z.object({
+  subjectLine: z.string().nullable(),
+  content: z.string(),
+});
+
+const $BaseTemplateInput = z.object({
+  name: z.string(),
+  type: z.enum(['NHS_APP', 'SMS', 'EMAIL', 'LETTER']),
+  version: z.number(),
+  fields: baseFields,
+  createdAt: z.string().optional(),
+});
+
+const $NHSAppTemplateInput = $BaseTemplateInput.extend({
+  type: z.literal('NHS_APP'),
+});
+
+const $EmailTemplateInput = $BaseTemplateInput.extend({
+  type: z.literal('EMAIL'),
+});
+
+const $SMSTemplateInput = $BaseTemplateInput.extend({
+  type: z.literal('SMS'),
+});
+
+const $LetterTemplateInput = $BaseTemplateInput.extend({
+  type: z.literal('LETTER'),
+});
+
+export const $TemplateInput = z.discriminatedUnion('type', [
+  $NHSAppTemplateInput,
+  $EmailTemplateInput,
+  $SMSTemplateInput,
+  $LetterTemplateInput,
+]);
+
+export const $Template = $TemplateInput.and(z.object({ id: z.string() }));
+
+export type TemplateInput = z.infer<typeof $TemplateInput>;
+export type Template = z.infer<typeof $Template>;
```

Comment:
this file has been added but is completely unused - maybe a rebase issue? can probably be deleted

Reply:
Removed